editor.planx.uk/src/ui/shared/DataTable/utils.tsx

```diff
@@ -0,0 +1,91 @@
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import { ValueOptions } from "@mui/x-data-grid";
+import React from "react";
+
+import { False, True } from "./components/cellIcons";
+import { DoubleInputFilter } from "./components/DoubleInputFilter";
+import { SingleInputFilter } from "./components/SingleInputFilter";
+import { ColumnRenderType, ColumnType } from "./types";
+
+const isValidFilterInput = (filterItem: Record<string, any>): boolean => {
+  return (
+    Array.isArray(filterItem.value) &&
+    filterItem.value.length === 2 &&
+    filterItem.value.every((val) => val != null)
+  );
+};
+
+const containsItem = (item: string, value: string) => {
+  return item.toLowerCase().includes(value.toLowerCase());
+};
+
+export const createFilterOperator = (
+  columnValueOptions: ValueOptions[],
+  filteringOne: boolean,
+) => [
+  {
+    value: "contains",
+    getApplyFilterFn: (filterItem: Record<string, any>) => {
+      if (filteringOne) {
+        return getSingleFilterFn(filterItem);
+      }
+      return getDoubleFilterFn(filterItem);
+    },
+    InputComponent: filteringOne ? SingleInputFilter : DoubleInputFilter,
```

Comment:
Ideally, it seems like this would/should be select-one or select-many (rather than select-two). Out of curiousity, have you tried passing in our existing `SelectMultiple` component here?